modules/@apostrophecms/piece-type/index.js

```diff
@@ -938,6 +953,28 @@ module.exports = {
             throw self.apos.error('invalid', 'Document has neither slug nor title, giving up');
           }
         }
+      },
+
+      async getDraftToShare(req) {
+        const { _id } = req.params;
+
+        if (!_id) {
+          throw self.apos.error('invalid');
+        }
+
+        const piece = await self.findOneForEditing(req, {
+          _id
+        });
+
+        if (!piece) {
+          throw self.apos.error('notfound');
+        }
+
+        if (piece.aposMode !== 'draft') {
+          throw self.apos.error('invalid');
```

Comment:
will this allow someone without the appropriate permissions to discover that a piece exists?

Reply:
That's right, was in the specs but maybe that a notfound would be enough here. User doesn't need more infos.